File: cs470/proj1/Tennis/SetScore.cpp
// Changes made by Zachary Thompson

#include <iostream>
#include <iomanip>
#include <cstdlib>
#include "SetScore.hpp"

SetScore::SetScore( Player *p1, Player *p2 ): Score(p1, p2), tieScore(0) {}
  
bool SetScore::haveAWinner() { 
    return ( ((p1Score >= 6 || p2Score >= 6) && abs(p1Score - p2Score) >= 2) ||
            tieScore != NULL );
}
bool SetScore::shouldPlayATieBreaker() { 
    return p1Score == 6 && p2Score == 6;
}

void SetScore::addTieScore( TieBreakerScore *score ) {
    addScore( score->getWinner() );
    this->tieScore = score;
}

void SetScore::print() {
    std::cout << std::setw(10) << player1Score() << std::setw(18) << player2Score();
    if( tieScore != NULL )
      tieScore->print();
    std::cout << std::endl;
}

File: cs470/proj1/ObjectiveCTennis/TennisInObjectiveC/SetScore.m
//  Additions made by Zachary Thompson
//  SetScore.m
//  TennisInObjectiveC
//
//  Created by student on 2/3/14.
//  Copyright (c) 2014 Ali Kooshesh. All rights reserved.
//

#import "SetScore.h"

@interface SetScore ()

@property (nonatomic) TieBreakerScore *tbscore;

@end

@implementation SetScore

-(BOOL) haveAWinner
{
    return ( ((self.player1Score >= 6 || self.player2Score >= 6) && abs( self.player1Score - self.player2Score ) >= 2) || self.tbscore);
}

-(BOOL) shouldPlayTieBreaker
{
    return self.player1Score == 6 && self.player2Score == 6;
}

-(void) addTieScore:(TieBreakerScore *)tbs {
    [self addScore:[tbs winner]];
    self.tbscore = tbs;
}

-(NSString *) description
{
    NSMutableString *result = [NSMutableString stringWithFormat:@"%10d%18d", self.player1Score, self.player2Score];
    
    if (self.tbscore) {
        [result appendString:[self.tbscore description]];
    }
    return result;
}

@end
